File: solutions/javascript/armstrong-numbers/1/armstrong-numbers.js

export function isArmstrongNumber(number) {

  const digits = number.toString().split('').map(d => Number.parseInt(d));
  const exponent = digits.length
  const powerSum = digits.map(d => d**exponent).reduce((acc,val) => acc+val, 0);
  
  return number == powerSum;
};

File: solutions/dart/armstrong-numbers/1/lib/armstrong_numbers.dart
final _values = {
  '0': BigInt.from(0),
  '1': BigInt.from(1),
  '2': BigInt.from(2),
  '3': BigInt.from(3),
  '4': BigInt.from(4),
  '5': BigInt.from(5),
  '6': BigInt.from(6),
  '7': BigInt.from(7),
  '8': BigInt.from(8),
  '9': BigInt.from(9)
};

final _zero = BigInt.from(0);

class ArmstrongNumbers {
  bool isArmstrongNumber(String number) {
    var sum = BigInt.from(0);
    int power = number.length;
    for (int i = number.length - 1; i >= 0; i--) {
      sum += (_values[number[i]] ?? _zero).pow(power);
    }
    return "${sum.toString()}" == number;
  }
}
